lib/ansible/modules/windows/win_domain_object_info.py

```diff
@@ -0,0 +1,121 @@
+short_description: retrieve  info regarding domain objects of all kinds
+description:
+   - With the module you can retrieve and output detailed information about all kinds of AD objects.
+options:
+  filter:
+    description:
+      - Powershell style filter.
+    type: str
+  identity:
+    description:
+      - Identity parameter used to find the Object in Active Directory.
+      - This value can be in the forms C(Distinguished Name), C(objectGUID).
+    type: str
+  include_deleted_objects:
+    description:
+      - Should deleted objects be included.
+    type: bool
+    default: false
+  ldap_filter:
+    description:
+      - LDAP style filter.
+    type: str
+  search_scope:
+    description:
+      - The scope of the search.
+      - A Base query searches only the current path or object.
+      - A OneLevel query searches the immediate children of that path or object.
+      - A Subtree query searches the current path or object and all children of that path or object.
+    type: str
+    default: Subtree
+    choices:
+      - Base
+      - OneLevel
+      - Subtree
+  search_base:
+    description:
+      - Where in the Active Directory should the search begin.
+    type: str
+  json_depth:
+    description:
+      - The number of levels into the object properties that should be parsed to JSON.
+    type: int
+    default: 3
+  properties:
+    description:
+      - A list of properties to retrieve.
+    type: list
+    elements: str
+    default: "*"
+  username:
+    description:
+      - The username to use when interacting with AD.
+      - If this is not set then the user Ansible used to log in with will be
+        used instead when using CredSSP or Kerberos with credential delegation.
+    type: str
+    aliases:
+      - domain_username
+  password:
+    description:
+      - The password for I(username).
+    type: str
+    aliases:
+      - domain_password
+  server:
+    description:
+      - Specifies the Active Directory Domain Services instance to connect to.
+      - Can be in the form of an FQDN or NetBIOS name.
+      - If not specified then the value is based on the domain of the computer
+        running PowerShell.
+    type: str
+    aliases:
+      - domain_server
+author:
+  - Shachaf Goldstein (@Shachaf92)
+'''
+
+EXAMPLES = r'''
+- name: Get ALL objects
+  win_domain_object_info:
+    filter: "*"
+
+- name: Get by filter
+  win_domain_object_info:
+    filter: "SAMAccountName -eq 'test*'"
+  register: filtered_result
+
+- name: Output first result
+  debug:
+    var: filtered_result.objects[0]
+
+- name: Get by filter in a specific OU
+  win_domain_object_info:
+    filter: "SAMAccountName -eq 'test*'"
+    searchBase: "OU=Users,DC=contoso,DC=com"
+  register: filtered_result_ou
+
+- name: Output first result
+  debug:
+    var: filtered_result_ou.objects[0]
+'''
+
+RETURN = r'''
+objects:
+    description: List of retrieve d objects, each formatted to JSON according to the I(jsonDepth) option.
```

Comment:
Although I think the `json_depth` option should be renamed as per my comment earlier.

```suggestion
    description: List of retrieved objects, each formatted at the maximum depth set by I(json_depth).
```